autogpt_platform/backend/backend/data/integrations.py

```diff
@@ -54,14 +54,33 @@ def from_db(webhook: IntegrationWebhook):
             config=dict(webhook.config),
             secret=webhook.secret,
             provider_webhook_id=webhook.providerWebhookId,
-            attached_nodes=(
-                [NodeModel.from_db(node) for node in webhook.AgentNodes]
-                if webhook.AgentNodes is not None
-                else None
-            ),
         )
 
 
+class WebhookWithRelations(Webhook):
+    triggered_nodes: list[NodeModel]
+    triggered_presets: list[LibraryAgentPreset]
+
+    @staticmethod
+    def from_db(webhook: IntegrationWebhook):
+        if webhook.AgentNodes is None or webhook.AgentPresets is None:
+            raise ValueError(
+                "AgentNodes and AgentPresets must be included in "
+                "IntegrationWebhook query with relations"
+            )
+        return WebhookWithRelations(
+            **Webhook.from_db(webhook).model_dump(),
+            triggered_nodes=[NodeModel.from_db(node) for node in webhook.AgentNodes],
+            triggered_presets=[
+                LibraryAgentPreset.from_db(preset) for preset in webhook.AgentPresets
+            ],
+        )
+
+
+# Fix Webhook <- NodeModel relations
```

Comment:
Any clarity on what the fix would be?

Reply:
The line below is the fix, this isn't a TODO :)